src/Rings/FractionalIdeal.jl

```diff
@@ -0,0 +1,41 @@
+export fractional_ideal
+
+import Nemo: numerator, denominator, iszero
+
+mutable struct FractionalIdeal{S <: Ideal, T <: RingElem}
+  num::S
+  den::T
+end
+
+function fractional_ideal(n::Ideal, d::RingElem)
```

Comment:
Should check `base_ring(n) === parent(d)`.

Reply:
ok. Is there any reason for
```julia
julia> Oscar.AbstractAlgebra.numerator === Oscar.Nemo.numerator
false
```

Reply:
No, open an issue in Nemo.